### ISSUE TITLE
"Target not supported" errors for Android build (API 19)

### ISSUE THREAD

**Issue by wps132230** [awslabs/aws-c-common#275]
When I am building AWS C++ SDK with Android NDK (API 19)
It gives me "error: use of undeclared identifier 'SIZE_MAX'", so I update the dependency: aws-c-common to its latest version. Then it gives me the following errors:
```
In file included from /aws-sdk-cpp/build/.deps/build/src/AwsCCommon/include/aws/common/array_list.h:19:
/aws-sdk-cpp/build/.deps/build/src/AwsCCommon/include/aws/common/math.h:112:6: error: "Target not supported"
#    error "Target not supported"
     ^
/aws-sdk-cpp/build/.deps/build/src/AwsCCommon/include/aws/common/math.h:126:6: error: "Target not supported"
#    error "Target not supported"
     ^
/aws-sdk-cpp/build/.deps/build/src/AwsCCommon/include/aws/common/math.h:139:6: error: "Target not supported"
#    error "Target not supported"
     ^
/aws-sdk-cpp/build/.deps/build/src/AwsCCommon/include/aws/common/math.h:154:6: error: "Target not supported"
#    error "Target not supported"
```
The cmake I am running to build C++ SDK:
```
cmake -DMINIMIZE_SIZE=ON -DANDROID_NATIVE_API_LEVEL=19 -DBUILD_ONLY=core -DCUSTOM_MEMORY_MANAGEMENT=1 -DENABLE_RTTI=OFF -DSTATIC_LINKING=1 -DCMAKE_BUILD_TYPE=Release -DSIMPLE_INSTALL=OFF -DGENERATE_VERSION_INFO=0 -DMINIMIZE_SIZE=ON -DTARGET_ARCH=ANDROID ..
```
I can see these errors only for API 19, and API 21 works fine for me.

**Comment by justinboswell** [comment]
Interesting, so android isn't defining SIZE_MAX at all? Do you happen to know if it's in a non-standard header or something? I'll do some research.

**Comment by justinboswell** [comment]
Oh, looks like we need to include limits.h for android. I'll have a PR up shortly.

**Comment by justinboswell** [comment]
Can you try pulling https://github.com/awslabs/aws-c-common/pull/276 and see if it fixes your issue?

**Comment by justinboswell** [comment]
I am able to successfully compile for android with the head revision of master with the following cmake configuration (the fix from #276 above was merged):

```bash
git clone https://github.com/awslabs/aws-c-common.git
mkdir -p aws-c-common/build
cd aws-c-common/build
cmake -DCMAKE_BUILD_TYPE=Release -DTARGET_ARCH=ANDROID -DCMAKE_TOOLCHAIN_FILE=/opt/android-ndk/build/cmake/android.toolchain.cmake -DANDROID_NATIVE_API_LEVEL=19 -DANDROID_TOOLCHAIN=clang -DANDROID_NDK=/opt/android-ndk -DANDROID_ABI=arm64-v8a ..
make
```

**Comment by justinboswell** [comment]
I have committed android CI to master, it will run on all builds now. We are using ndk-r19c with NATIVE_API_LEVEL=19. If you are still experiencing issues, let me know!
